streaming/src/main/scala/io/gearpump/streaming/source/DataSource.scala

```diff
@@ -29,10 +29,14 @@ import io.gearpump.{TimeStamp, Message}
  * {{{
  *  GenStringSource extends DataSource {
  *
- *    def open(context: TaskContext, startTime: Option[TimeStamp]): Unit = {}
+ *    def open(context: TaskContext, startTime: TimeStamp): Unit = {}
  *
- *    def read(batchSize: Int): List[Message] = {
- *      List.fill(batchSize)(Message("message"))
+ *    def advance(): Boolean = {
```

Comment:
I don't get why we need this API?

Can a read() automatically step forward the cursor? If yes, maybe a better naming function for this is "hasData()".